### PR TITLE
Improved documentation and tests about input and output functions

### PR DESCRIPTION
**Pull request by mvsusp**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1377,7 +1377,7 @@ An example of ``output_fn`` for the accept type "application/python-pickle" can
 
     import numpy as np
 
-    def output_fn(data, accepts):
+    def output_fn(prediction_result, accepts):
         """An output_fn that dumps a pickled numpy as response"""
         if request_content_type == "application/python-pickle":
             return np.dumps(data)
```
Comment (laurenyu) [inline README.rst]: should `data` be changed to `prediction_result` here?

File: README.rst [hunk 2]
```diff
@@ -1377,7 +1377,7 @@ An example of ``output_fn`` for the accept type "application/python-pickle" can
 
     import numpy as np
 
-    def output_fn(data, accepts):
+    def output_fn(prediction_result, accepts):
         """An output_fn that dumps a pickled numpy as response"""
```
Comment (laurenyu) [inline README.rst]: change 'numpy' to 'object' for consistency with the above change?

File: README.rst [hunk 3]
```diff
@@ -1386,6 +1386,9 @@ An example of ``output_fn`` for the accept type "application/python-pickle" can
             # if the content type is not supported.
             pass  
 
+A example with the ``input_fn`` and ``output_fn`` above can be find in
```
Comment (laurenyu) [inline README.rst]: two small changes:
* remove 'the'
* s/find in/found

File: tests/data/cifar_10/source/resnet_cifar_10.py [hunk 4]
```diff
@@ -106,21 +108,19 @@ def model_fn(features, labels, mode, params):
...
 
-def eval_input_fn(training_dir, hyperpameters):
-    return input_fn(tf.estimator.ModeKeys.EVAL,
-                    batch_size=BATCH_SIZE, data_dir=training_dir)
+def eval_input_fn(training_dir, hyperparameters):
+    return _input_fn(tf.estimator.ModeKeys.EVAL, batch_size=BATCH_SIZE)
 
 
-def input_fn(mode, batch_size, data_dir):
+def _input_fn(mode, batch_size):
```
Comment (laurenyu) [inline tests/data/cifar_10/source/resnet_cifar_10.py]: I think it'd be better if this method's name were a little more distinct from `input_fn`

File: tests/data/cifar_10/source/resnet_cifar_10.py [hunk 5]
```diff
@@ -138,3 +138,12 @@ def input_fn(mode, batch_size, data_dir):
     labels = tf.contrib.framework.local_variable(labels, name='labels')
 
     return {INPUT_TENSOR_NAME: images}, labels
+
+
+def input_fn(serialized_data, content_type):
+    data = pickle.loads(serialized_data)
+    return data
```
Comment (laurenyu) [inline tests/data/cifar_10/source/resnet_cifar_10.py]: you could make this a one-line method: `return pickle.loads(serialized_data)`

File: tests/integ/test_tf_cifar.py [hunk 6]
```diff
@@ -19,12 +22,34 @@
 from tests.integ import DATA_DIR, REGION
 from tests.integ.timeout import timeout_and_delete_endpoint, timeout
 
+PICKLE_CONTENT_TYPE = "application/python-pickle"
```
Comment (laurenyu) [inline tests/integ/test_tf_cifar.py]: use single quotes instead of double quotes for consistency with the rest of the SDK

File: README.rst (hunk 1 above)
Comment (mvsusp) [reply README.rst]: I did that for consistency, look at the code block below [here](https://github.com/mvsusp/sagemaker-python-sdk/blob/8de091b4dad1023cdf34a9a5e5edf7341e7ee342/README.rst#endpoint-prediction-request-handling)

File: README.rst (hunk 2 above)
Comment (mvsusp) [reply README.rst]: sure

File: README.rst (hunk 3 above)
Comment (mvsusp) [reply README.rst]: ok

File: tests/data/cifar_10/source/resnet_cifar_10.py (hunk 4 above)
Comment (mvsusp) [reply tests/data/cifar_10/source/resnet_cifar_10.py]: good point.

File: README.rst (hunk 1 above)
Comment (laurenyu) [reply README.rst]: I meant for line 1383 - sorry for the confusion

Comment (mvsusp) [reply README.rst]: ok

**Comment by laurenyu** [comment]
also, don't forget imperative mood with the overall commit message :)

**Pull request merged**